packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -125,11 +125,17 @@ const helpers = {
         },
         httpContent
       );
+    } else {
+      // no httpContent found, returning empty body
+      return {
+        code,
+        mediaType: 'text/plain',
+        example: {
+          externalValue: '',
+          key: 'default',
+        },
+      };
```

Comment:
First of all, the key 'default' is ignored in further processing (it's just forced by type).

Questions:
1) there is a content but it's not found by negotiation and not `application/json`
a) return any content (pick first)
b) return default (text/plain '')
c) throw error

2) there is no content at all
a) return default (text/plain '')
b) throw error

For me, for both it feels more natural to just throw errors (the way it was implemented).

However from user perspective it might be more handy to return empty text/plain. If so, I'd return it for both cases to keep consistent (like it's implemented now). User can start the server faster with less effort and problems.

@philsturgeon - curious your opinion

Reply:
I am not sure what the intention of the example, or externalValue are doing here. 

When there is literally no response defined in OpenAPI, we still need to respond:

```
openapi: 3.0.1
info:
  title: API
  version: 1.0.0
paths:
  /:
    get:
      responses:
        '200':
          description: OK
```

I was not really trying to suggest we change how defaults work, as I am not sure how the specific code should be written. The feature request was just to make sure this example responds without failing.

Our mime type logic does seem a bit funny. We should be looking for one that matches the `Accept` header, if that is not there we should be looking for `default`, if that is not there we should _technically_ error. 

However that is probably not very useful for most of our users, who may well expect it to use JSON by default (as so many people use JSON without ever even checking Accept). So, having application/json as another check cant hurt. If you put Accept: application/xml it should certainly return you XML.

But when you define nothing in responses, you should get nothing. That's what I'd expect. "Nothing" is defined as text/plan with an empty string.

Reply:
> I am not sure what the intention of the example, or externalValue are doing here.
> 
> When there is literally no response defined in OpenAPI, we still need to respond:
> 
> ```
> openapi: 3.0.1
> info:
>   title: API
>   version: 1.0.0
> paths:
>   /:
>     get:
>       responses:
>         '200':
>           description: OK
> ```
> 
> I was not really trying to suggest we change how defaults work, as I am not sure how the specific code should be written. The feature request was just to make sure this example responds without failing.
> 
> Our mime type logic does seem a bit funny. We should be looking for one that matches the `Accept` header, if that is not there we should be looking for `default`, if that is not there we should _technically_ error.
> 
> However that is probably not very useful for most of our users, who may well expect it to use JSON by default (as so many people use JSON without ever even checking Accept). So, having application/json as another check cant hurt. If you put Accept: application/xml it should certainly return you XML.
> 
> But when you define nothing in responses, you should get nothing. That's what I'd expect. "Nothing" is defined as text/plan with an empty string.

💯, content negotiation for the win!

Reply:
Wrapping it all up:
- try to match any existing content type before we we fallback to `text/plan`
- support `Accept` header
- if no mediaType is matched generate `text/plain` ""
- make `example` optional. Returning "default" example looks like a hack and is relying on internal knowledge of the implementation detail. Tight coupling.
- if neither `example` nor `schema` is defined generate empty message.

This looks like a candidate for a separate PR and separate story (IIRC SO-80 isn't about media types anyway but thought that is a small change so didn't mention it before)

@philsturgeon @StefanDywersant sounds good?

Reply:
> make example optional. Returning "default" example looks like a hack and is relying on internal knowledge of the implementation detail. Tight coupling.
The "default" is not an internal knowledge. It's just a label because all examples need to have label. It's not being used anywhere. I can change it to "Krzysztof" and it will work ;)

> if neither example nor schema is defined generate empty message.
Btw. that's the reason I'm using example. text/plain has no schema.

@chris-miaskowski  If I understand correctly you suggesting to do all those in separate story?

Reply:
yes

Reply:
awesome summary on all points @chris-miaskowski, different JIRA issues for each of them, and can we shove in some tests to make that clear its whats happening. :)

Reply:
@StefanDywersant can you pls create a story and move those changes to a separate branch/pr pls?
and add a reference to that new story somewhere in this PR, cool?

Reply:
Without above code this story is not done. It will throw error when you try a request with almost empty spec.

However I will create a new story for those things we've been talking about - Accept, content negotiation etc.

Are you ok with that?

Reply:
Hmm, I thought this story was about `servers` not about undefined responses? Where am I wrong? Are the two things related somehow?

Reply:
Yes they are. There were two barricades to cross. First when there was no servers defined, second when there was no proper response (which was actual jira issue).

Like I said, I'd leave it in this PR and create another one which extends that behaviour with actually looking into Accept: and available responses. Extend, not change. Empty response will stay for one of the cases.

Reply:
All right! Sounds good!

Reply:
I can cope with this — although there's another step we should remember — OAS2 defines a global [produces](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md) key where there are defaults; these should be respected in case nothing is defined.

This change, though, should be done in graphite and somehow it should arrive here.

Reply:
@XVincentX Well yeah that's actually a very good point!

Now thinking about this a bit more I'm leaning towards a bit different solution.

This whole thing should IMO be **validation**. E.g. if I request "text/plain", but such media type cannot be produced we should return 4xx.

@philsturgeon WDYT?

Reply:
The problem here is that this global property does not exist in OAS3; in that case, we're back to the original issue. 

Therefore I think what we can do is to use that property in OAS2 documents (and so Graphite shall somehow bring it here), otherwise, we'll have to agree on something (and I'm ok with `text/plain`)

Reply:
Hm, I'd extend a bit that idea of returning text/plain.

If we are at the point where we need to return default response, then let's try to match the response with Accept header. We could prepare several types of empty responses:
- text/plain: ''
- application/json: null
- application/xml: ''

How about that?

Reply:
I created task for Accept header and refining the flow of picking example https://stoplightio.atlassian.net/browse/SO-98